framework/src/media/utils/MediaUtils.cpp

```diff
@@ -68,10 +72,257 @@ audio_type_t getAudioTypeFromPath(std::string datapath)
+		*sample_rate = 24000;
+		break;
+	case 7 :
+		*sample_rate = 22050;
+		break;
+	case 8 :
+		*sample_rate = 16000;
+		break;
+	case 9 :
+		*sample_rate = 12000;
+		break;
+	case 10 :
+		*sample_rate = 11025;
+		break;
+	case 11 :
+		*sample_rate = 8000;
+		break;
+	case 12 :
+		*sample_rate = 7350;
+		break;
+	default :
+		medvdbg("sample rate is wrong\n");
+		return false;
+	}
+	bit = header[2];
+	bit <<= 2;
+	bit |= (header[3]>>6);
+	bit &= 7;
+	switch (bit) {
+	case 0 :
+		medvdbg("channel is wrong\n");
+		return false;
+	case 1 :
+		*channel = 1;
+		break;
+	case 2 :
+		*channel = 2;
+		break;
+	case 3 :
+		*channel = 3;
+		break;
+	case 4 :
+		*channel = 4;
+		break;
+	case 5 :
+		*channel = 5;
+		break;
+	case 6 :
+		*channel = 6;
+		break;
+	case 7 :
+		*channel = 8;
+		break;
+	default :
+		medvdbg("channel is wrong\n");
+		return false;
+	}
+
+	return true;
+}
+
+bool wave_header_parsing(unsigned char *header, unsigned int *channel, unsigned int *sample_rate, audio_format_type_t *pcmFormat) //header 44byte
+{
+	*channel = header[22];
+
+	*sample_rate = header[25];
+	*sample_rate <<= 8;
+	*sample_rate |= header[24];
+
+	unsigned short bit_per_sample = 0;
+
+	bit_per_sample |= header[35];
+	bit_per_sample <<= 8;
+	bit_per_sample |= header[34];
+
+	switch (bit_per_sample) {
+	case 8 :
+		*pcmFormat = AUDIO_FORMAT_TYPE_S8;
+		break;
+	case 16 :
+		*pcmFormat = AUDIO_FORMAT_TYPE_S16_LE;
+		break;
+	case 32 :
+		*pcmFormat = AUDIO_FORMAT_TYPE_S32_LE;
+		break;
+	default :
+		medvdbg("bit per sample is wrong.\n");
+		return false;	
+	}
+
+	return true;
+}
+
+bool header_parsing(FILE *fp, audio_type_t audio_type, unsigned int *channel, unsigned int *sample_rate, audio_format_type_t *pcmFormat)
+{
+	unsigned char *header;
+	unsigned char tag[2];
+	bool is_header;
+
+	switch (audio_type) {
+	case AUDIO_TYPE_MP3 :
+		is_header = false;
+		while (fread(tag, sizeof(unsigned char), 1, fp) > 0) {
+			if (tag[0] == 0xFF) {
+				is_header = true;
+				break;
+			}
+		}
+		if (is_header) {
+			header = (unsigned char *)malloc(sizeof(unsigned char) * MP3_HEADER_LENGTH);
+			fseek(fp, -1, SEEK_CUR);
+			fread(header, sizeof(unsigned char), MP3_HEADER_LENGTH, fp);
+			if (!mp3_header_parsing(header, channel, sample_rate, pcmFormat)) {
+				free(header);
+				return false;
+			}
+		}
+		else {
+			medvdbg("no header\n");
+			return false;
+		}
+		break;
+	case AUDIO_TYPE_AAC :
+		header = (unsigned char *)malloc(sizeof(unsigned char) * AAC_HEADER_LENGTH);
+		fread(header, sizeof(unsigned char), AAC_HEADER_LENGTH, fp);
+		if (!aac_header_parsing(header, channel, sample_rate, pcmFormat)) {
+			free(header);
+			return false;
+		}
+		break;
+	case AUDIO_TYPE_WAVE :
+		header = (unsigned char *)malloc(sizeof(unsigned char) * WAVE_HEADER_LENGTH);
+		fread(header, sizeof(unsigned char), WAVE_HEADER_LENGTH, fp);
+		if (!wave_header_parsing(header, channel, sample_rate, pcmFormat)) {
+			free(header);
+			return false;
+		}
+		break;
+	case AUDIO_TYPE_PCM :
```

Comment:
not good approach, datasource has its own default setting or get info from user.
I will update comment in open() function right after.